hack/ginkgo-e2e.sh

```diff
@@ -152,7 +159,17 @@ CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-${KUBE_CONTAINER_RUNTIME:-}}
 # Add path for things like running kubectl binary.
 PATH=$(dirname "${e2e_test}"):"${PATH}"
 export PATH
-"${ginkgo}" "${ginkgo_args[@]:+${ginkgo_args[@]}}" "${e2e_test}" -- \
+
+# Choose the program to execute.
+program=("${ginkgo}")
+if [[ "${E2E_TEST_DEBUG_TOOL}" == "delve" ]]; then
+  program=("dlv" "exec")
+fi
+if [[ "${E2E_TEST_DEBUG_TOOL}" == "gdb" ]]; then
+  program=("gdb")
+fi
+
+"${program[@]:+${program[@]}}" "${ginkgo_args[@]:+${ginkgo_args[@]}}" "${e2e_test}" -- \
```

Comment:
Part of me wants to say the first + is unnecessary here. We expect program to always be set by this point. It makes more sense for the args, as those may or may not be set

Reply:
Good observation, I've removed the expansion of `program` because it's never null